hal/src/photon/socket_hal.cpp

```diff
@@ -978,3 +978,17 @@ sock_handle_t socket_handle_invalid()
 {
     return SOCKET_INVALID;
 }
+
+sock_result_t socket_join_multicast(const HAL_IPAddress *address, network_interface_t /*nif*/, void * /*reserved*/)
+{
+    wiced_ip_address_t multicast_address;
+    SET_IPV4_ADDRESS(multicast_address, address->ipv4);
+    return as_sock_result(wiced_multicast_join(WICED_STA_INTERFACE, &multicast_address));
```

Comment:
Okay, I just didn't know, if the nif converts 1-to-1 from hal to wiced. I did not find a place elsewhere where it is used. And it's slightly underspecified, I think...